src/Controller/PostsController.php

```diff
@@ -47,16 +52,15 @@ public function articlePage(
     ) {
         $articleRepository = $this->getDoctrine()->getRepository(Post::class)->setPosts('article');
         $article = $articleRepository->findBySlug($slug);
-        // @TODO: redirect to 404 page
         if (null === $article) {
-            return $this->index($manager);
+            throw new NotFoundHttpException('La page demandée n\'existe pas');
```

Comment:
Pro tip : use **altgr + G** to print real apostrophe (and avoid slash usage) : `’`